packages/javascript/bh-shared-ui/src/views/SSOConfiguration/SSOConfiguration.tsx

```diff
@@ -0,0 +1,275 @@
+// Copyright 2024 Specter Ops, Inc.
+//
+// Licensed under the Apache License, Version 2.0
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+import { useState, Dispatch, FC, useCallback, useMemo, ChangeEvent } from 'react';
+import { faSearch } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { Box, Grid, Typography, Paper, TextField, useTheme } from '@mui/material';
+import { useMutation, useQuery } from 'react-query';
+import { CreateOIDCProvideRequest, SSOProvider } from 'js-client-library';
+import { apiClient } from '../../utils';
+import {
+    CreateSAMLProviderDialog,
+    CreateSAMLProviderFormInputs,
+    CreateMenu,
+    ConfirmationDialog,
+    DocumentationLinks,
+    SSOProviderTable,
+    PageWithTitle,
+    SSOProviderInfoPanel,
+} from '../../components';
+import CreateOIDCProviderDialog from '../../components/CreateOIDCProviderDialog';
+import { useFeatureFlag } from '../../hooks';
+
+const MakeSSOConfiguration = async (
+    addSnackbar: (notification: string, key: string, options?: any) => void,
+    useAppDispatch: () => Dispatch<any>
+): Promise<{ default: FC }> => ({
+    default: () => {
```

Comment:
I havent seen this pattern before and it feels like react wouldnt know what to do with this MakeSSOConfiguration function. Looks like it would be a component because theres some markdown, but it returns an object with this default key which seems like a hook. 

Do you mind explaining the intention behind this?

Reply:
It's a component factory essentially but this particular component is lazy loaded by both BHCE and BHE. The lazy loader expected the shape. 🤷

Reply:
Hell yes it is!🧑‍🍳 🤌  
That came out clean 🧼